client/verta/tests/unit_tests/test_deployed_model.py

```diff
@@ -456,26 +461,84 @@ def test_batch_predict_with_batches_and_indexes(mocked_responses) -> None:
     """ Since the input has 5 rows and we're providing a batch_size of 1, we expect 5 batches.
     Include an example of an index.
     """
-    expected_df_list = [pd.DataFrame({"A": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
-                       pd.DataFrame({"B": [11, 12, 13, 14, 15, 16, 17, 18, 19, 20]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
-                       pd.DataFrame({"C": [21, 22, 23, 24, 25, 26, 27, 28, 29, 30]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
-                       pd.DataFrame({"D": [31, 32, 33, 34, 35, 36, 37, 38, 39, 40]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
-                       pd.DataFrame({"E": [41, 42, 43, 44, 45, 46, 47, 48, 49, 50]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
-                       ]
+    expected_df_list = [
+        pd.DataFrame({"A": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}, index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        pd.DataFrame({"B": [11, 12, 13, 14, 15, 16, 17, 18, 19, 20]},
+                     index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        pd.DataFrame({"C": [21, 22, 23, 24, 25, 26, 27, 28, 29, 30]},
+                     index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        pd.DataFrame({"D": [31, 32, 33, 34, 35, 36, 37, 38, 39, 40]},
+                     index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        pd.DataFrame({"E": [41, 42, 43, 44, 45, 46, 47, 48, 49, 50]},
+                     index=["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]),
+        ]
     for expected_df in expected_df_list:
         mocked_responses.add(
             responses.POST,
             BATCH_PREDICTION_URL,
             body=json.dumps(expected_df.to_dict(orient="split")),
             status=200,
-            )
+        )
     creds = EmailCredentials.load_from_os_env()
     dm = DeployedModel(
         prediction_url=PREDICTION_URL,
         creds=creds,
         token=TOKEN,
-        )
+    )
     input_df = pd.DataFrame({"a": [1, 2, 3, 4, 5], "b": [11, 12, 13, 14, 15]}, index=["A", "B", "C", "D", "E"])
     prediction_df = dm.batch_predict(input_df, 1)
     expected_final_df = pd.concat(expected_df_list)
     pd.testing.assert_frame_equal(expected_final_df, prediction_df)
+
+
+@st.composite
+def generate_data(draw):
+    """ Return a dict that represents a dataframe. Generates ints, floats, and strings."""
+    num_rows = draw(st.integers(min_value=50, max_value=200))
```

Comment:
Is there a reason not to go with

```suggestion
    num_rows = draw(st.integers(min_value=1, max_value=200))
```

? I would imagine we'd want our functionality to work with DataFrames down to 1 row.